Attempt to fix console. (#4294)

tomusdrw opened the pull request:
* Attempt to fix console.
* Fixing couple of possible exceptions

Closes #4290

jacogr commented on js/src/dapps/static/console.js:
```diff
@@ -17,15 +17,19 @@
+	    	return;
+	    }
+	    refreshWatches();
+	});
```
Indentation of this block looks very random, like mixing on tabs/spaces. (Pre-linting, visual check)

jacogr commented on js/src/dapps/static/console.js:
```diff
@@ -17,15 +17,19 @@
-    }
-    refreshWatches();
-  });
+if (typeof(window.parent.secureApi) == 'object')
```
Should probably be `===` (along with the one below)

jacogr commented on js/src/dapps/static/console.js:
```diff
@@ -56,21 +60,42 @@ function evaluate(x) {
     return `"<span class="${typeof(x)}Type">${escapeHtml(x)}</span>"`;
-  if (Object.prototype.toString.call(x) === '[object Array]')
-    return `[${x.map(displayReady).join(', ')}]`;
+  if (toString.call(x) === '[object Array]')
```
`Array.isArray` ?

jacogr commented on js/src/dapps/static/console.js:
```diff
@@ -159,7 +184,7 @@ function exec() {
       try {
         res = evaluate(c);
-        if (typeof(res) == 'object' && res !== null && res.constructor.name == "Promise") {
+        if (typeof(res) == 'object' && res !== null && res.constructor && res.constructor.name == "Promise") {
```
`===`

Be aware that `res.constructor.name` may not work in FF with the es6-promise polyfills. (Have not checked myself with this branch, but Promise checking this way is always a bit of an issue)

jacogr commented:
Some minor grumbles.

tomusdrw commented:
I've noticed some other issues with console app (it works in development mode, but fails in release). Will prepare a fix for master first (addressing those issues) and then re-open this.
